lib/widgets/components/chip_component.dart

```diff
@@ -0,0 +1,123 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_internship/helpers/spacer_width.dart';
+
+enum ChipType { Flat, Clean }
+
+class ChipComponent extends StatelessWidget {
+  final ChipType chipType;
+  final IconData? icon;
+  final String label;
+
+  const ChipComponent({
+    Key? key,
+    required this.chipType,
+    this.icon,
+    required this.label,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    switch (chipType) {
+      case ChipType.Flat:
+        return FlatChip(label: label, icon: icon);
+      case ChipType.Clean:
+        return CleanChip(label: label, icon: icon);
+    }
```

Comment:
There is new way of writing switch case in Flutter in new version of Flutter. Try it.